tests/s3/copy_object/test_copy_object_tag.py

```diff
@@ -0,0 +1,1102 @@
+                                                  self.object_name1,
+                                                  self.file_path,
+                                                  mb_count=self.mb_count,
+                                                  key=self.key_src,
+                                                  value=self.value_src)
+        self.log.info("Step 1: Uploaded an object and tag is set for object")
+        self.log.debug("Retrieving tag of an object %s", self.object_name1)
+        put_resp = self.tag_obj.get_object_tags(self.bucket_name1, self.object_name1)
+        assert_utils.assert_true(put_resp[0], put_resp[1])
+        self.log.debug("Retrieved tag of an object")
+        self.log.info("Step 2: Copy object to same bucket with different object.")
+        status, response = self.s3_test_obj.copy_object(self.bucket_name1,
+                                                        self.object_name1,
+                                                        self.bucket_name1,
+                                                        self.object_name2,
+                                                        TaggingDirective='REPLACE',
+                                                        Tagging=tag_str)
+        assert_utils.assert_true(status, response)
+        copy_etag = response['CopyObjectResult']['ETag']
+        self.log.info("Step 3: Retrieving tag of a destination object %s", self.object_name2)
+        copy_resp = self.tag_obj.get_object_tags(self.bucket_name1, self.object_name2)
+        assert_utils.assert_true(copy_resp[0], copy_resp[1])
+        self.log.info("Step 3: Retrieved tag of an object")
+        self.log.info("Step 4: Compare tagset of source and destination object")
+        assert_utils.assert_not_equal(copy_resp[1][0]["Key"], put_resp[1][0]["Key"],
+                                      put_resp[1][0]["Key"])
+        assert_utils.assert_not_equal(copy_resp[1][0]["Value"], put_resp[1][0]["Value"],
+                                      put_resp[1][0]["Value"])
+        self.log.info("Step 4: Compared and verified tags of source and destination object"
+                      " are different")
+        self.log.info("Verify destination tag set match with assigned tag set in copy object")
+        assert_utils.assert_equal(copy_resp[1][0]["Key"], self.key_dest, self.key_dest)
+        assert_utils.assert_equal(copy_resp[1][0]["Value"], self.value_dest, self.value_dest)
+        self.log.info("Step : Verification of etag and metadata for data integrity check")
+        self.copy_obj_di_check(self.bucket_name1, self.object_name1, self.bucket_name1,
+                               self.object_name2, put_etag=put_etag, copy_etag=copy_etag)
+        self.log.info("Step : Verification of etag and metadata for data integrity check")
+        self.log.info("Step 5: Copy-object with destination tag set same as source object"
+                      "in the different bucket")
```

Comment:
the tag string is not same i beleive , as we are doing replace here.Pls modify this step string

Reply:
Updated